src/module/httprequest.py

```diff
@@ -0,0 +1,123 @@
+# pylint: disable=missing-module-docstring
+# pylint: disable=dangerous-default-value
+from typing import Any
+from typing import Dict
+
+import requests
+from requests.models import Response
+from src.constants.messages import BASIC_AUTH_PARAMS_MISSING
+from src.constants.messages import MISSING_AUTH_TOKEN
+from src.constants.messages import UNKNOWN_AUTH_TYPE
+from src.constants.str import AUTHORIZATION
+from src.constants.str import BASIC_AUTH
+from src.constants.str import BEARER_TOKEN
+from src.constants.str import HEADERS
+from src.constants.str import NO_AUTH
+from src.module.config import Config
+from src.module.headers import Header
+from src.module.url import URL
+from src.utils.helper import set_request_session
+
+
+class HttpRequest:
+    """
+    It handles all configuration, validation and API requests.
+    """
+
+    def __init__(self, base_url: str, auth_type: str = NO_AUTH, **kwargs):
+        """
+        Initialize and configure the HtppRequest.
+
+        :param base_url: base url for all api request
+        :param kwargs: other configurations required for http request.
+        """
+        self.url = URL(base_url)
+        self.auth_type = auth_type
+
+        if self.auth_type not in [NO_AUTH, BASIC_AUTH, BEARER_TOKEN]:
+            raise ValueError(UNKNOWN_AUTH_TYPE)
+
+        if HEADERS in kwargs.keys():
+            self.headers = Header(kwargs[HEADERS])
+            del kwargs[HEADERS]
+        else:
+            self.headers = Header({})
+
+        self.config = Config(**kwargs)
+        self.session = requests.Session()
+        self._http = set_request_session(self.config)
+
+    @property
+    def http(self) -> requests.Session:
+        """Get http request session object.
+
+        :return: Request session object
+        :rtype: requests.Session
+        """
+        return self._http
+
+    @http.setter
+    def http(self, http: requests.Session):
+        """Set http/https session object.
+
+        :param http: Request session object
+        :type http: requests.Session
+        """
+        self._http = http
+
+    def _validate_auth(self, headers, **kwargs):
+        """
+        Validate the auth if its mandatory for request.
```

Comment:
```suggestion
        """Validate the auth if its mandatory for request.
```

DocString should always start at the first line.